RustLeanModels/Basic.lean

```diff
@@ -13,19 +12,23 @@ abbrev Str := List Char
 
 --LOGIC
 
-lemma exists_minima_prop_nat {p: Nat → Prop} : (∃ x , p x) → ∃ m, (p m) ∧ (∀ n, p n → m ≤ n) :=by
+theorem exists_minima_prop_nat {p: Nat → Prop} : (∃ x , p x) → ∃ m, (p m) ∧ (∀ n, p n → m ≤ n) :=by
   intro gn; obtain ⟨n, gn⟩ := gn
-  induction n using Nat.strong_induction_on
+  induction n using Nat.strongInductionOn
   rename_i n ind
   by_cases (∀ m < n, ¬ p m ); rename_i gm
```

Comment:
Small simplification here to: you can give a name to the `by_case ...` to avoid having `rename_i` twice in each branch:
```
    by_cases gm: ∀ m < n, ¬ p m
```

Reply:
TIL! Done.